moveit_experimental/moveit_jog_arm/src/jog_arm.cpp

```diff
@@ -0,0 +1,426 @@
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/*      Title     : jog_arm.cpp
+ *      Project   : moveit_jog_arm
+ *      Created   : 3/9/2017
+ *      Author    : Brian O'Neil, Andy Zelenak, Blake Anderson
+ */
+
+#include "moveit_jog_arm/jog_arm.h"
+
+static const std::string LOGNAME = "jog_arm";
+
+namespace moveit_jog_arm
+{
+JogArm::JogArm(ros::NodeHandle& nh, const planning_scene_monitor::PlanningSceneMonitorPtr& planning_scene_monitor)
+  : nh_(nh), planning_scene_monitor_(planning_scene_monitor)
+{
+  // Read ROS parameters, typically from YAML file
+  if (!readParameters())
+    exit(EXIT_FAILURE);
+
+  // loop period
+  period_ = ros::Duration(parameters_.publish_period);
+
+  // Publish freshly-calculated joints to the robot.
+  // Put the outgoing msg in the right format (trajectory_msgs/JointTrajectory or std_msgs/Float64MultiArray).
+  if (parameters_.command_out_type == "trajectory_msgs/JointTrajectory")
+    outgoing_cmd_pub_ = nh_.advertise<trajectory_msgs::JointTrajectory>(parameters_.command_out_topic, 1);
+  else if (parameters_.command_out_type == "std_msgs/Float64MultiArray")
+    outgoing_cmd_pub_ = nh_.advertise<std_msgs::Float64MultiArray>(parameters_.command_out_topic, 1);
+
+  // publish commands (for c++ interface)
+  twist_stamped_pub_ = nh_.advertise<geometry_msgs::TwistStamped>(parameters_.cartesian_command_in_topic, 1);
+  joint_jog_pub_ = nh_.advertise<control_msgs::JointJog>(parameters_.joint_command_in_topic, 1);
+
+  // ROS Server for allowing drift in some dimensions
+  drift_dimensions_server_ =
+      nh_.advertiseService(nh_.getNamespace() + "/" + ros::this_node::getName() + "/change_drift_dimensions",
+                           &JogArm::changeDriftDimensions, this);
+
+  // ROS Server for changing the control dimensions
+  dims_server_ =
```

Comment:
This could be called `control_dimensions_server_`.

Reply:
I agree and like this change